fenix/app/src/main/java/org/mozilla/fenix/components/metrics/MetricsMiddleware.kt

```diff
@@ -20,18 +24,39 @@ class MetricsMiddleware(
         next: (AppAction) -> Unit,
         action: AppAction,
     ) {
-        handleAction(action)
+        handleAction(context, action)
         next(action)
     }
 
-    private fun handleAction(action: AppAction) = when (action) {
+    private fun handleAction(context: MiddlewareContext<AppState, AppAction>, action: AppAction) = when (action) {
         is AppAction.AppLifecycleAction.ResumeAction -> {
             metrics.track(Event.GrowthData.SetAsDefault)
             metrics.track(Event.GrowthData.FirstAppOpenForDay)
             metrics.track(Event.GrowthData.FirstWeekSeriesActivity)
             metrics.track(Event.GrowthData.UsageThreshold)
             metrics.track(Event.GrowthData.UserActivated(fromSearch = false))
         }
+
+        is AppAction.AwesomeBarAction.EngagementFinished -> {
+            val interactions = context.state.awesomeBarState.fxSuggestPingsInfo()
+            interactions.forEach {
+                when (it.interactionInfo) {
+                    is FxSuggestInteractionInfo.Amp -> {
```

Comment:
We only added impressions and clicks for AMP (sponsored) suggestions for now, but we'll want to add them for other types of suggestions eventually—so we expect this `when` will grow arms for `FxSuggestInteractionInfo.Wikipedia`, `Amo`, `Pocket`, etc. in the future, and that they'll have different fields to report in different metrics.